docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+The class diagram for the storage module is as seen below: 
+
+
+![](images/storageclass.PNG)
+
+&nbsp;
+
+On startup, Duke invokes the loadSavedAppointment() and loadSavedPatient() methods in Storage. This allows the program 
+to retrieve previously stored data from a .txt file and convert it into the static AppointmentList and PatientList objects for use
+within the program. 
+
+The Storage object creates a Scanner object that will parse individual lines in the .txt file, convert them into
+new Appointments, and then add them to an ArrayList called `appointmentListToReturn`. This `appointmentListToReturn` will be passed back to Duke to
+construct the static AppointmentList. The sequence diagram is shown below:
+
+![](images/loadsavedappt_seq.PNG)
+![](images/loadsavedappt_ref1.PNG)
+
+When the static AppointmentList or PatientList has changes, or the program is exiting, saveAppointmentList() or savePatientList() 
+is invoked respectively. This allows the Storage object to back up existing records to a local .txt file.
+
+The Storage object will create a FileWriter object called `fw`. The command will then iterate through the existing AppointmentList
+and parse each Appointment within, converting it to a string. `fw` then writes this string to the .txt file.
+The sequence diagram is shown below:
+
+![](images/saveapptlist_seq.PNG)
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.4 A&D command module 
+
+The command module consist of 11 different classes, where each class does a different command by itself. 
+These classes allows the patients and appointments to be added into HAMS, allows the updating of patient and 
+appointment details, allows the admin to delete patients and appointment information and allows of listing of the 
+different patients and the appointments in HAMS. In addition, the classes also deals with displaying the list of 
+commands and as well as to allowing HAMS to exit.
+
+![](images/Command%20class%20diagram.png)
+
+All of these command classes inherits from the abstract ```Command``` class. Likewise, the execute function of each 
+command  class is also inherited from the abstract ```Command``` class's ```execute(Ui ui, Storage storage)``` function 
+too. Every command class (other than the  ```ExitCommand``` and ```HelpCommand```) are actually façade classes that 
+creates the connection from the ```Main``` class to the other classes required such as ```Storage``` class,
+```PatientList``` class and the ```Appointment``` class to name a few. 
+
+##### 2.2.4.1 AddPatientClass
+
+To add a patient, the ```AddPatientCommand``` class is used. For this ```AddPatientCommand``` class, it serves as the 
+façade class for the ```Main```, ```Patient``` , ```PatientList``` and the ```Storage``` class to interact with one 
+another. Also, to uniquely identify a patient, an unique patientId number is assigned to each patient when they are first added into the patient list.
+
+![](images/AddPatientDiagram.png)
```

Comment:
Same problem here: consider omitting some details (e.g. private variables) to make the diagram less complicated?